File: backend/app/models/workspace.py
from sqlalchemy import String, Integer, ForeignKey, Boolean, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class Workspace(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "workspaces"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    slug: Mapped[str] = mapped_column(String(50), unique=True, nullable=False)
    logo_url: Mapped[str | None] = mapped_column(String(800), nullable=True)
    organization_size: Mapped[str | None] = mapped_column(String(50), nullable=True)
    timezone: Mapped[str] = mapped_column(String(255), default="UTC")
    owner_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("users.id"), nullable=False)
    
    owner: Mapped["User"] = relationship(foreign_keys=[owner_id])
    members: Mapped[list["WorkspaceMember"]] = relationship(back_populates="workspace")
    projects: Mapped[list["Project"]] = relationship(back_populates="workspace")

class WorkspaceMember(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "workspace_members"
    
    workspace_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("workspaces.id"), nullable=False)
    user_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("users.id"), nullable=False)
    role: Mapped[int] = mapped_column(Integer, default=15)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    
    workspace: Mapped["Workspace"] = relationship(back_populates="members")
    user: Mapped["User"] = relationship(back_populates="workspaces")

File: backend-go/internal/model/workspace.go
package model

// Workspace represents a team workspace.
type Workspace struct {
	BaseModel

	Name             string  `gorm:"size:255;not null" json:"name"`
	Slug             string  `gorm:"uniqueIndex;size:50;not null" json:"slug"`
	LogoURL          *string `gorm:"size:800" json:"logo_url"`
	OrganizationSize *string `gorm:"size:50" json:"organization_size"`
	Timezone         string  `gorm:"size:255;default:UTC" json:"timezone"`
	OwnerID          uint64  `gorm:"not null" json:"owner_id"`

	// Relationships
	Owner    User              `gorm:"foreignKey:OwnerID" json:"-"`
	Members  []WorkspaceMember `gorm:"foreignKey:WorkspaceID" json:"-"`
	Projects []Project         `gorm:"foreignKey:WorkspaceID" json:"-"`
}

func (Workspace) TableName() string {
	return "workspaces"
}

// WorkspaceMember represents a user's membership in a workspace.
type WorkspaceMember struct {
	BaseModel

	WorkspaceID uint64 `gorm:"not null;uniqueIndex:idx_ws_member_user" json:"workspace_id"`
	UserID      uint64 `gorm:"not null;uniqueIndex:idx_ws_member_user" json:"user_id"`
	Role        int    `gorm:"default:15" json:"role"`
	IsActive    bool   `gorm:"default:true" json:"is_active"`

	// Relationships
	Workspace Workspace `gorm:"foreignKey:WorkspaceID" json:"-"`
	User      User      `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

func (WorkspaceMember) TableName() string {
	return "workspace_members"
}
